File: server/src/main/java/de/pnp/manager/model/item/Jewellery.java
package de.pnp.manager.model.item;

public class Jewellery extends Equipment implements IJewellery {

    private String gem;

    public Jewellery() {
        super();
        this.gem = "";
    }

    @Override
    protected boolean shouldBreak() {
        return false;
    }

    @Override
    public Jewellery copy() {
        Jewellery jewellery = (Jewellery) super.copy();
        jewellery.setGem(this.getGem());

        return jewellery;
    }

    public String getGem() {
        return gem;
    }

    public void setGem(String gem) {
        this.gem = gem;
    }

    @Override
    public boolean equals(Object o) {

        if (!(o instanceof Jewellery) || !super.equals(o)) {
            return false;
        }

        Jewellery other = (Jewellery) o;

        return this.getGem().equals(other.getGem());
    }
}

File: app/src/main/java/de/pnp/manager/app/model/Jewellery.kt
package de.pnp.manager.app.model

import de.pnp.manager.model.ICurrency
import de.pnp.manager.model.IRarity
import de.pnp.manager.model.item.IEquipment
import de.pnp.manager.model.item.IJewellery
import de.pnp.manager.model.upgrade.IUpgrade
import java.util.ArrayList

class Jewellery: IJewellery, Equipment() {
    private var gem : String = ""

    override fun copy(): IJewellery {
        TODO("Not yet implemented")
    }

    override fun getGem(): String {
        return gem
    }

    override fun setGem(gem: String) {
        this.gem = gem
    }

}